examples/feature_flags_example/server/plugin.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+} from '@kbn/core/server';
+import { combineLatest } from 'rxjs';
+
+import { defineRoutes } from './routes';
+
+export class FeatureFlagsExamplePlugin implements Plugin {
+  private readonly logger: Logger;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+  }
+
+  public setup(core: CoreSetup) {
+    const router = core.http.createRouter();
+
+    // Register server side APIs
+    defineRoutes(router);
+  }
+
+  public start(core: CoreStart) {
+    // Promise form: when we need to fetch it once, like in an HTTP request
+    void Promise.all([
+      core.featureFlags.getBooleanValue('example-boolean', false),
```

Comment:
Thanks! I'm also wondering if there's any valid use case where code tries to fetch a flag that isn't registered for some reason?

Reply:
Technically, the flags don't need to be registered to run.

That's why I am trying to decouple the gitops flow (the one requiring flags to be programmatically registered) and the actual runtime. I'm hoping this would avoid unnecessarily having extra objects in memory during runtime (to keep the registry).

> any valid use case where code tries to fetch a flag that isn't registered

I can only think about 2 edge scenarios:
1. Registered by another plugin: similar to UI Settings, another plugin of the same solution might register the flag, and it can be reused in other related plugins/packages.
2. We need to create a flag manually (outside of our gitops process) for whatever reason (that type of flag is not supported and we really-really-really need it).

Reply:
Makes sense to me, thanks for clarifying!